tests/main/views/test_buyers.py

```diff
@@ -3098,3 +3144,328 @@ def test_do_not_show_question_and_answer_dates_for_closed_brief(self, data_api_c
+            "briefResponses": {
+                "id": 5678, "supplierName": "BananaCorp",
+                "awardDetails": {"pending": True}
+            }
+        }
+
+    def teardown_method(self, method):
+        self.data_api_client_patch.stop()
+        super(TestAwardBriefDetails, self).teardown_method(method)
+
+    def test_award_brief_details_200s_with_correct_default_content(self):
+        self.login_as_buyer()
+        res = self.client.get(self.url.format(brief_id=1234, brief_response_id=5678))
+
+        assert res.status_code == 200
+        document = html.fromstring(res.get_data(as_text=True))
+        self.assert_breadcrumbs(res, extra_breadcrumbs=[
+            (
+                'I need a thing to do a thing',
+                '/buyers/frameworks/digital-outcomes-and-specialists-2/requirements/digital-outcomes/1234'
+            )
+        ])
+
+        page_title = self._strip_whitespace(document.xpath('//h1')[0].text_content())
+        assert page_title == "TellusaboutyourcontractwithBananaCorp"
+
+        submit_button = document.xpath('//input[@class="button-save" and @value="Submit"]')
+        assert len(submit_button) == 1
+
+        secondary_link_text = document.xpath('//div[@class="secondary-action-link"]//a[1]')[0]
+        assert secondary_link_text.text_content() == "Back to previous page"
+
+        secondary_link = document.xpath('//div[@class="secondary-action-link"]//a[1]/@href')[0]
+        assert secondary_link == \
+            '/buyers/frameworks/digital-outcomes-and-specialists-2/requirements/digital-outcomes/1234/award-contract'
+
+    def test_award_brief_details_post_valid_form_calls_api_and_redirects(self):
+        self.data_api_client.update_brief_award_details.return_value = api_stubs.framework(
+            slug='digital-outcomes-and-specialists-2',
+            status='awarded',
+            lots=[
+                api_stubs.lot(slug='digital-outcomes', allows_brief=True)
+            ]
+        )
+        with self.app.app_context():
+            self.login_as_buyer()
+            res = self.client.post(
+                self.url.format(brief_id=1234, brief_response_id=5678),
+                data={
+                    "awardedContractStartDate-day": "31",
+                    "awardedContractStartDate-month": "12",
+                    "awardedContractStartDate-year": "2020",
+                    "awardedContractValue": "88.84"
+                }
+            )
+
+            assert self.data_api_client.update_brief_award_details.call_args == mock.call(
+                '1234', '5678',
+                {'awardedContractStartDate': "2020-12-31", "awardedContractValue": "88.84"},
+                updated_by="buyer@email.com"
+            )
+            assert res.status_code == 302
+            assert res.location == "http://localhost/buyers"
+            self.assert_flashes("updated-brief")
+
+    def _setup_api_error_response(self, error_json):
```

Comment:
I don't have a strong opinion on this but I tend to keep helpers near the tests that use them. However in this case the class isn't that big so it doesn't really make a difference 😄 I'll move it underneath the setup/teardown.